cocos/physics/physx/character-controllers/physx-character-controller.ts

```diff
@@ -0,0 +1,268 @@
+        } else {
+            return true;
+        }
+    }
+
+    onEnable (): void {
+        this._isEnabled = true;
+        if (!this._impl) {
+            this.onComponentSet();
+        }
+        this.setDetectCollisions(true);
+        this.setOverlapRecovery(true);
+        (PhysicsSystem.instance.physicsWorld as PhysXWorld).addCCT(this);
+    }
+
+    onDisable (): void {
+        this._isEnabled = false;
+        this.onDestroy();
+    }
+
+    onLoad (): void {
+
+    }
+
+    release (): void {
+        if (this._impl) {
+            if (this._impl.$$) {
+                PX.IMPL_PTR[this._impl.$$.ptr] = null;
+                delete PX.IMPL_PTR[this._impl.$$.ptr];
+            }
+            this._impl.release();
+            this._impl = null;
+        }
+    }
+
+    onDestroy (): void {
+        this.release();
+
+        (PhysicsSystem.instance.physicsWorld as PhysXWorld).removeCCT(this);
+    }
+
+    getPosition (out: IVec3Like): void {
+        if (!this._impl) return;
+        Vec3.copy(out, this._impl.getPosition());
+    }
+
+    setPosition (value: IVec3Like): void {
+        if (!this._impl) return;
+        this._impl.setPosition(value);
+    }
+
+    setContactOffset (value: number): void {
+        if (!this._impl) return;
+        this._impl.setContactOffset(value);
+    }
+
+    setStepOffset (value: number): void {
+        if (!this._impl) return;
+        this._impl.setStepOffset(value);
+    }
+
+    setSlopeLimit (value: number): void {
+        if (!this._impl) return;
+        this._impl.setSlopeLimit(Math.cos(degreesToRadians(value)));
+    }
+
+    setDetectCollisions (value: boolean): void {
+        if (!this._impl) return;
+        this._impl.setCollision(value);
+    }
+
+    setQuery (value: boolean): void {
+        if (!this._impl) return;
+        this._impl.setQuery(value);
+    }
+
+    setOverlapRecovery (value: boolean): void {
+        this._overlapRecovery = value;
+    }
+
+    onGround (): boolean {
+        return (this._pxCollisionFlags & (1 << 2)) > 0;//PxControllerCollisionFlag::Enum::eCOLLISION_DOWN
+    }
+
+    syncSceneToPhysics (): void {
+        const node = this.characterController.node;
+        if (node.hasChangedFlags) {
+            //only sync scale from scene node to physics
+            if (node.hasChangedFlags & TransformBit.SCALE) this.syncScale();
+        }
+    }
+
+    syncPhysicsToScene (): void {
+        this.getPosition(v3_0);
+        v3_0.subtract(this._comp.scaledCenter);
+        this._comp.node.setWorldPosition(v3_0);
+    }
+
+    syncScale () {
+        this.updateScale();
+    }
+
+    // eNONE = 0,   //!< the query should ignore this shape
+    // eTOUCH = 1,  //!< a hit on the shape touches the intersection geometry of the query but does not block it
+    // eBLOCK = 2   //!< a hit on the shape blocks the query (does not block overlap queries)
+    queryCallback = {
+        preFilter (filterData: any, shape: any, _actor: any, _out: any): number {
```

Comment:
Seem likes it can be a static method instead instance method.

Reply:
yes